pytorch_forecasting/models/base_model.py

```diff
@@ -1012,7 +1013,7 @@ def plot_prediction(
             # move to cpu
             y = y.detach().cpu()
             # create figure
-            if ax is None:
+            if (ax is None) or (not ax_provided):
```

Comment:
this is confusing. I would say: `ax_not_provided = ax is None` before the loop and then `if ax_not_provided` after the loop